### ISSUE TITLE
Disable search auto-update, when change page 

### ISSUE THREAD

**Issue by thethir13en** [opensearch-project/OpenSearch-Dashboards#1549]
Hello!

I've noticed non-intuitive behavior, while exploring logs with "Refresh every N time" option enabled: 
when I'm switching from first page to other one in "Discover" window, each refresh iteration returns me to
the first page of search. 

I think, it will be more intuitive, if page switching will disable auto-refresh option. Because at the moment of 
search process it's really annoying, when you are waiting for exact event, skipping it, switching page and Opensearch dashboards
returns you back to the first page because of "Refresh every N time" enabled.

**Comment by boktorbb-amzn** [comment]
Hello @thethir13en , could you possibly give some more detailed reproduction steps for this behavior?

**Comment by joshuarrrr** [comment]
@ananzh This seems like a good observations about the conflict between auto-refresh and basic pagination. Do we also see something similar with pagination in the new table vis?

**Comment by ananzh** [comment]
@joshuarrrr  I think this is a good point. In table vis, we didn't implement any interaction with auto fresh. So even auto fresh is on, pagination won't get refreshed.

**Comment by joshuarrrr** [comment]
> So even auto fresh is on, pagination won't get refreshed.

That's not true, was just able to repro on playground. The problem is that the autorefresh actually reloads each embeddable, so unless pagination state is saved on the dashboard somehow it will be lost.

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/1679762/74ce0a62-8742-4f60-bf9f-3fbaf58fca54

@dagneyb @wbeckler this does seem like quite an annoyance for embedded tables on dashboards, even though the original issue is for discover.

**Comment by ananzh** [comment]
@joshuarrrr I mean in visualization. In table vis, it won't trigger refresh. Actually I think this is same in other visualizations. I just re-check an area graph `Flight Count and Average Ticket Price` which has no auto refresh. 

Currently, only discover and dashboards have which as you shown. I think when you asked for `new table vis` you mean table visualization right?
